src/block-management/block-manager.js

```diff
@@ -275,15 +275,11 @@ export class BlockManager extends React.Component<PropsType, StateType> {
 		return ( content === undefined || content === '' ) && ( innerBlocks.length === 0 );
 	}
 
-	isCandidateForReplaceBlock( block: BlockType ) {
-		return ( block.name === 'core/paragraph' || block.name === 'core/heading' || block.name === 'core/code' );
-	}
-
 	isReplaceable( block: ?BlockType ) {
 		if ( ! block ) {
 			return false;
 		}
-		return this.isEmptyBlock( block ) && this.isCandidateForReplaceBlock( block );
+		return this.isEmptyBlock( block ) && isUnmodifiedDefaultBlock( block );
```

Comment:
I think you don't need the empty block check, since `isUnmodifiedDefaultBlock` already checks if the block has the default attributes

Reply:
addressed in 49e8630 - I still decided to keep `isReplaceable()` as a method because it makes it easier to read and follow the code where the method is called.